wpilib/wpilib/pidbase.py

```diff
@@ -0,0 +1,501 @@
+# validated: 2018-01-27 DS 738a1c015cfb edu/wpi/first/wpilibj/PIDBase.java
+#----------------------------------------------------------------------------
+# Copyright (c) FIRST 2008-2018. All Rights Reserved.
+# Open Source Software - may be modified and shared by FRC teams. The code
+# must be accompanied by the FIRST BSD license file in the root directory of
+# the project.
+#----------------------------------------------------------------------------
+
+import threading
+import warnings
+
+import hal
+
+from .interfaces import PIDSource
+from .sendablebase import SendableBase
+from .resource import Resource
+from .timer import Timer
+from .lineardigitalfilter import LinearDigitalFilter
+
+__all__ = ["PIDBase"]
+
+class PIDBase(SendableBase):
+    """Class implements a PID Control Loop.
+
+    Creates a separate thread which reads the given PIDSource and takes care of the integral
```

Comment:
Heh, this seems wrong; this class doesn't spawn a thread. But it's what upstream has. I guess we should file a bug, I suppose.

Reply:
my impression was it leaves the choice of notifier vs non notifier up to the subclass

Reply:
I think that's the intention here, yes. But that doesn't change the fact that there's a documentation bug here.